p2p/transport/webrtc/sdp.go

```diff
@@ -0,0 +1,121 @@
+package libp2pwebrtc
+
+import (
+	"crypto"
+	"fmt"
+	"net"
+
+	"github.com/multiformats/go-multihash"
+)
+
+const clientSDP string = `
+v=0
+o=- 0 0 IN %s %s
+s=-
+c=IN %s %s
+t=0 0
+m=application %d UDP/DTLS/SCTP webrtc-datachannel
+a=mid:0
+a=ice-options:trickle
+a=ice-ufrag:%s
+a=ice-pwd:%s
+a=fingerprint:sha-256 ba:78:16:bf:8f:01:cf:ea:41:41:40:de:5d:ae:22:23:b0:03:61:a3:96:17:7a:9c:b4:10:ff:61:f2:00:15:ad
+a=setup:actpass
+a=sctp-port:5000
+a=max-message-size:16384
+`
+
+func renderClientSdp(addr *net.UDPAddr, ufrag string) string {
+	ipVersion := "IP4"
+	if addr.IP.To4() == nil {
+		ipVersion = "IP6"
+	}
+	return fmt.Sprintf(
+		clientSDP,
+		ipVersion,
+		addr.IP,
+		ipVersion,
+		addr.IP,
+		addr.Port,
+		ufrag,
+		ufrag,
+	)
+}
+
+const serverSDP string = `
```

Comment:
For maintainability purposes I would seriously recommend to turn this into a template string or the like, where you assign stuf based on variables, as the amount of `%s` here is staggering high, very easy to make a mistake here.

Reply:
There's a `text/template` package in the standard library. Not sure how it compares performance-wise, might be worth micro-benchmarking before committing.

Reply:
I won't be replacing this with a template because there is a large difference in performance:
```
goos: linux
goarch: amd64
pkg: github.com/libp2p/go-libp2p/p2p/transport/webrtc
cpu: AMD Ryzen 9 5950X 16-Core Processor            
BenchmarkRenderClientSDP
BenchmarkRenderClientSDP-32              1284057               975.3 ns/op           552 B/op         10 allocs/op
BenchmarkRenderClientSDPTemplate
BenchmarkRenderClientSDPTemplate-32        89923             12512 ns/op            2819 B/op         75 allocs/op
PASS
ok      github.com/libp2p/go-libp2p/p2p/transport/webrtc        3.462s
```

Reply:
Fair enough. It's a shame that they do not allow named format args in Go, I guess because syntax wise how would that even work in Go.

At the very least make them index parameters (https://pkg.go.dev/fmt#hdr-Explicit_argument_indexes) so that you at least have some way to reuse and visually see what parameters are used multiple times :)

Reply:
Looks fine now, please feel free to resolve (I do not seem to have the powers to resolve it myself).